tinker/static/assets/js/tinker.js

```diff
@@ -14,4 +14,117 @@ $(document).ready(function () {
+
+        var totalPages = Math.ceil(numberOfItems / limitPerPage);
+
+        createButtons(totalPages, limitPerPage);
+
+        var currentPage = 1;
+
+        var grandTotal = limitPerPage * currentPage;
+
+        for (var i = grandTotal - limitPerPage; i < grandTotal ; i ++) {
+            $("#loop .list-group:eq(" + i + ")").show();
+        }
+
+        switchPages(limitPerPage);
+
+        nextPage(totalPages, limitPerPage);
+
+    });
+
+    var totalPages = Math.ceil(numberOfItems / limitPerPage);
+
+    createButtons(totalPages, limitPerPage);
+
+    switchPages(limitPerPage);
+
+    nextPage(totalPages, limitPerPage);
+
+    previousPage(totalPages, limitPerPage);
+}
+
+function nextPage(totalPages, limitPerPage) {
+    $("#next-page").on("click", function() {
+        var currentPage = $(".pagination li.active").index();
+        if (currentPage === totalPages) {
+            return false;
+        } else {
+            currentPage ++;
+            $(".pagination li").removeClass("active");
+            $("#loop .list-group").hide();
+
+            var grandTotal = limitPerPage * currentPage;
+
+            for (var i = grandTotal - limitPerPage; i < grandTotal ; i ++) {
+                $("#loop .list-group:eq(" + i + ")").show();
+            }
+            $(".pagination li.current-page:eq(" + (currentPage - 1) + ")").addClass("active");
+        }
+    });
+}
+
+function previousPage(totalPages, limitPerPage) {
+    $("#previous-page").on("click", function() {
+        var currentPage = $(".pagination li.active").index();
+        if (currentPage === 1) {
+             return false;
+        } else {
+             currentPage --;
+             $(".pagination li").removeClass('active');
+             $("#loop .list-group").hide();
+
+             var grandTotal = limitPerPage * currentPage;
+
+             for (var i = grandTotal - limitPerPage; i < grandTotal; i++) {
+                 $("#loop .list-group:eq(" + i + ")").show();
+             }
+             $(".pagination li.current-page:eq(" + (currentPage - 1) + ")").addClass('active');
+         }
+    });
+}
+
+function switchPages(limitPerPage) {
+    $(".pagination li.current-page").on("click", function() {
+        if ($(this).hasClass("active")) {
+            return false;
+        } else {
+            var currentPage = $(this).index();
+            $(".pagination li").removeClass("active");
+            $(this).addClass("active");
+            $("#loop .list-group").hide();
+
+            var grandTotal = limitPerPage * currentPage;
+
+            for (var i = grandTotal - limitPerPage; i < grandTotal ; i ++) {
+                $("#loop .list-group:eq(" + i + ")").show();
+            }
+        }
+    });
+}
+
+function createButtons(totalPages, limitPerPage) {
+    $(" #loop .list-group:gt(" + (limitPerPage - 1) + ")").hide();
+
+    $(".pagination").append("<li class='current-page active to-remove'><a href='javascript:void(0)'>" + 1 + "</a></li>");
```

Comment:
normally using JS to create HTML is sad and should be avoided. But I think this is fine, since it keeps everything contained nicely in the methods.

Reply:
although looking deeper into this commit, you added the "select" with options as well. So i'm confused?